### PR TITLE
Add new rule `no-valueless-arguments`

### LINKED ISSUE #1882: New rule: `no-valueless-arguments`
The template parser allows arguments (or attributes) without values:
```hbs
<ComponentName
  @testArg
/>
```

Which will be interpreted as:
```hbs
<ComponentName
  @testArg=""
/>
```

I noticed this behavior while refactoring code and [while describing it in the Discord](https://discord.com/channels/480462759797063690/485447409296736276/826030166559686706) it was suggested to lint against this, since it's more of a footgun than a feature.

It seems that the [AST produced by `ember-template-recast`](https://astexplorer.net/#/gist/089bdaf8a5d1ba324e76b301f8f77e96/6a73969d614767c5fc292eec9a17753b829453f3) adds an `isValueless` property to the `AttrNode` so I think that can be used to do the detection?

### PR DESCRIPTION
**Pull request by Windvis**
Closes #1882

### CODE REVIEW

**Comment by lifeart** [comment]
Hi @Windvis, any progress on it? Let me know if you need any help to get it working.

**Comment by Windvis** [comment]
@lifeart Hey! Thanks for pinging me. The only thing remaining is writing the documentation about the rule, but I'm not very good at it so I'm procrastinating :grimacing:. I'll try to pick it up this week.

**Comment by lifeart** [comment]
@Windvis you already have almost all for documentation) Just take original issue description, and put it to template, also, you may use as a reference - https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-arguments-for-html-elements.md

File: test/unit/rules/no-valueless-arguments-test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+generateRuleTests({
+  name: 'no-valueless-arguments',
+
+  config: true,
+
+  good: [
+    '<SomeComponent @emptyString="" data-test-some-component />',
+    `<button type="submit" disabled {{on "click" this.submit}}></button>`,
+  ],
```
Comment (bmish) [inline test/unit/rules/no-valueless-arguments-test.js:14]: Can we add some tests for curly components? `{{foo-component ... }}`

Comment (lifeart) [reply test/unit/rules/no-valueless-arguments-test.js:14]: @bmish I think it's good to land, for now we don't have AST logic (in this MR), related to curly stuff.

**Review by bmish — APPROVED** [review]
Thanks!

**Comment by lifeart** [comment]
@Windvis thank you for contribution!

**Pull request merged**